### ISSUE TITLE
Using newer libstdc++ with older glibc

### ISSUE THREAD

**Issue by serghov** [AppImageCrafters/appimage-builder#147]
I am running into issues when using newer gcc compiler to build c++ code on systems with older glibc versions.

For instance when building an application with `gcc 10.3` on `ubuntu 18.04`, appimage-builder bundles `glibc 2.27` from ubuntu, and `libstdc++.6` from my gcc 10.3 which is much newer. Later when trying to run the .appimage on `ubuntu 20.04`, the runtime detects that glibc version on the system is newer, and tries to use that, which is fine however it also tries to use `libstdc++.6` from the system, which is older than the one packaged (from `gcc 10.3`) and the application fails to run.

**Comment by azubieta** [comment]
Interesting situation, a quick fix would be to switch your build to ubuntu 20.04. But I guess that the runtime needs to be aware of the libstc++ version in order to pic the newer one as done with glibc.

**Comment by serghov** [comment]
@azubieta yes switching the build to ubuntu 20.04 fixes the issue, however the .appimage will not be forward compatible with future releases. 
I agree that AppRun should check libstdc++ version against the system one separately from glibc.
That would probably require keeping them in separate folders in AppDir.

**Comment by azubieta** [comment]
Interesting, could you please elaborate a bit more why it would not be forward compatible ?

**Comment by serghov** [comment]
Basically because of the same issue, if I compile and package the code on ubuntu 20.04, later if someone tries to run it on 21.04 or some other newer version, the runtime will choose to use glibc and libstdc++ from the system instead of packaged ones and may end up using an older libstdc++.

Actually why does the runtime need to choose between packaged and system libstdc++? Isn't using the packaged one always the best choice?

**Comment by azubieta** [comment]
@serghov the packages built using appimage-builder use certain system libraries such as graphics drivers, those may link to libstdc++ and if the system has a newer version than the one in the bundle they will fail to load. That's way we must check versions.

Could you please help me on finding a method of getting the libstc++ version that could be used from the AppRun?

**Comment by azubieta** [comment]
This will be addressed at https://github.com/AppImageCrafters/AppRun/issues/58
